plugin.py

```diff
@@ -36,6 +36,14 @@ def is_allowed_to_start(
         resolve_module_script = os.path.join(server_directory_path, 'resolve_module.js')
         first_folder = workspace_folders[0].path
         command =  [cls._node_bin(), resolve_module_script, first_folder, 'typescript/lib/tsserverlibrary.js']
-        workspace_ts_path = subprocess.check_output(command, universal_newlines=True)
+        startupinfo = None
+        # Prevent cmd.exe popup on Windows.
+        if os.name == "nt":
+            startupinfo = subprocess.STARTUPINFO()
+            startupinfo.vShowWindow = subprocess.SW_HIDE
+            startupinfo.dwFlags |= (
+                subprocess.STARTF_USESTDHANDLES | subprocess.STARTF_USESHOWWINDOW
+            )
```

Comment:
I don't believe that the `vShowWindow` property is valid in that structure. See the struct definition here https://docs.microsoft.com/en-us/windows/win32/api/processthreadsapi/ns-processthreadsapi-startupinfoa

We do this in other places in the code, for example here:

https://github.com/sublimelsp/LSP/blob/9f9ad2b48bc2160528688e3bef4eb991446ae029/plugin/core/transports.py#L291-L292

Would that be enough to use that?

Also, I'd check the platform using `sublime.platform() == "windows"`

Reply:
Especially wondering about that `STARTF_USESTDHANDLES`... Is that something that you have added on purpose?

Reply:
I'm not very experienced with Python/ST plugin development so I looked around and based my own approach on something similar to this one (iirc):

<https://github.com/bcomnes/sublime-standard-format/blob/39965154a3d6e33ffec596d7354e4e59e7be7647/standard-format.py#L296-L302>

Apparently I typed `vShowWindow` instead of `wShowWindow`, which indeed isn't valid 😬, however it's not required anyways it seems. I tested the approach you linked and it works just fine (Windows 10 + Fedora 35), so I updated the code accordingly.

Will double-check next time, thank you for your input!

Reply:
I would drop the unnecessary line breaks and parenthesis in the `dwFlags` assignment but it looks good now to me.